app/lib/common/widgets/share/action/share_space_object_action.dart

```diff
@@ -0,0 +1,161 @@
+import 'dart:io';
+
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/common/widgets/share/widgets/attach_options.dart';
+import 'package:acter/common/widgets/share/widgets/external_share_options.dart';
+import 'package:acter/common/widgets/share/widgets/file_share_options.dart';
+import 'package:acter/features/deep_linking/actions/show_qr_code.dart';
+import 'package:acter/features/deep_linking/types.dart';
+import 'package:acter/features/files/actions/download_file.dart';
+import 'package:acter/features/news/model/news_references_model.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:open_filex/open_filex.dart';
+import 'package:share_plus/share_plus.dart';
+
+typedef SpaceObjectDetails = ({
+  String spaceId,
+  ObjectType objectType,
+  String objectId,
+});
+
+typedef FileDetails = ({
+  File file,
+  String? mimeType,
+});
+
+Future<void> openShareSpaceObjectDialog({
+  required BuildContext context,
+  SpaceObjectDetails? spaceObjectDetails,
+  FileDetails? fileDetails,
+}) async {
+  await showModalBottomSheet(
+    showDragHandle: true,
+    useSafeArea: true,
+    context: context,
+    isScrollControlled: true,
+    isDismissible: true,
+    builder: (context) => ShareSpaceObjectActionUI(
+      spaceObjectDetails: spaceObjectDetails,
+      fileDetails: fileDetails,
+    ),
+  );
+}
+
+class ShareSpaceObjectActionUI extends StatelessWidget {
+  final SpaceObjectDetails? spaceObjectDetails;
+  final FileDetails? fileDetails;
+
+  const ShareSpaceObjectActionUI({
+    super.key,
+    this.spaceObjectDetails,
+    this.fileDetails,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return SingleChildScrollView(
+      child: Padding(
+        padding: const EdgeInsets.all(12),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          mainAxisAlignment: MainAxisAlignment.start,
+          children: [
+            if (spaceObjectDetails != null) ...[
+              attachmentOptionsUI(context),
+              SizedBox(height: 16),
+              externalShareOptionsUI(context),
+              SizedBox(height: 20),
+            ],
+            if (fileDetails != null) fileShareOptionsUI(context),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget attachmentOptionsUI(BuildContext context) {
+    String spaceId = spaceObjectDetails!.spaceId;
+    ObjectType objectType = spaceObjectDetails!.objectType;
+    String objectId = spaceObjectDetails!.objectId;
+
```

Comment:
hmm... I _know_ from the code that this is fine, but in the past `!` has bitten us in that when code around it is updated the assumption breaks and all we have is a rather undescriptive error in sentry. Can we refactor this to not have `!` here? e.g. by passing the object as a parameter?

Reply:
Yes sure.